security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -129,19 +129,22 @@ func (c *citadelClient) getTLSDialOption() (grpc.DialOption, error) {
 			return nil, fmt.Errorf("failed to append certificates")
 		}
 	}
-
-	config := tls.Config{}
-	config.RootCAs = certPool
-
-	if ProvCert != "" {
-		// Load the certificate from disk
-		certificate, err := tls.LoadX509KeyPair(ProvCert+"/cert-chain.pem", ProvCert+"/key.pem")
-		if err != nil {
-			return nil, fmt.Errorf("cannot load key pair: %s", err)
-		}
-
-		config.Certificates = []tls.Certificate{certificate}
+	var certificate tls.Certificate
+	config := tls.Config{
+		Certificates: []tls.Certificate{certificate},
+		GetClientCertificate: func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
+			log.Infof("call getclientCertificate")
+			if ProvCert != "" {
+				// Load the certificate from disk
+				certificate, err = tls.LoadX509KeyPair(ProvCert+"/cert-chain.pem", ProvCert+"/key.pem")
+				if err != nil {
+					return nil, fmt.Errorf("cannot load key pair: %s", err)
+				}
+			}
+			return &certificate, nil
```

Comment:
if provcert != nil, this will return `nil,nil`, is that what we want? same question on `Certificates: []tls.Certificate{certificate},`

Reply:
in other cases, we will use token instead of client cert.